jax_galsim/interpolatedimage.py

```diff
@@ -0,0 +1,1114 @@
+
+        flux_scaling *= image.scale**2
+
+        # Create an array of coordinates
+        coords = jnp.stack(image.get_pixel_centers(), axis=-1)
+        coords = coords * image.scale  # Scale by the image pixel scale
+        coords = coords - jnp.asarray(offset)  # Add the offset
+
+        # Apply the jacobian transformation
+        inv_jacobian = jnp.linalg.inv(jacobian)
+        _, logdet = jnp.linalg.slogdet(inv_jacobian)
+        coords = jnp.dot(coords, inv_jacobian.T)
+        flux_scaling *= jnp.exp(logdet)
+
+        im = _xValue_arr(
+            coords[..., 0],
+            coords[..., 1],
+            self._offset.x,
+            self._offset.y,
+            self._pad_image.bounds.xmin,
+            self._pad_image.bounds.ymin,
+            self._pad_image.array,
+            self._x_interpolant,
+        )
+
+        # Apply the flux scaling
+        im = (im * flux_scaling).astype(image.dtype)
+
+        # Return an image
+        return Image(array=im, bounds=image.bounds, wcs=image.wcs, check_bounds=False)
+
+    def _drawKImage(self, image, jac=None):
+        jacobian = jnp.eye(2) if jac is None else jac
+
+        # Create an array of coordinates
+        coords = jnp.stack(image.get_pixel_centers(), axis=-1)
+        coords = coords * image.scale  # Scale by the image pixel scale
+        coords = jnp.dot(coords, jacobian)
+
+        im = _kValue_arr(
+            coords[..., 0],
+            coords[..., 1],
+            self._offset.x,
+            self._offset.y,
+            self._kim.bounds.xmin,
+            self._kim.bounds.ymin,
+            self._kim.array,
+            self._kim.scale,
+            self._x_interpolant,
+            self._k_interpolant,
+        )
+        im = (im).astype(image.dtype)
+
+        # Return an image
+        return Image(array=im, bounds=image.bounds, wcs=image.wcs, check_bounds=False)
+
+
+@_wraps(_galsim._InterpolatedImage)
+def _InterpolatedImage(
+    image,
+    x_interpolant=Quintic(),
+    k_interpolant=Quintic(),
+    use_true_center=True,
+    offset=None,
+    gsparams=None,
+    force_stepk=0.0,
+    force_maxk=0.0,
+):
+    return InterpolatedImage(
+        image,
+        x_interpolant=x_interpolant,
+        k_interpolant=k_interpolant,
+        use_true_center=use_true_center,
+        offset=offset,
+        gsparams=gsparams,
+        calculate_maxk=False,
+        calculate_stepk=False,
+        pad_factor=1.0,
+        flux=jnp.sum(image.array),
+        _force_stepk=force_stepk,
+        _force_maxk=force_maxk,
+        _recenter_image=False,
+    )
+
+
+def _xValue_arr(x, y, x_offset, y_offset, xmin, ymin, arr, x_interpolant):
+    vals = _draw_with_interpolant_xval(
+        x + x_offset,
+        y + y_offset,
+        xmin,
+        ymin,
+        arr,
+        x_interpolant,
+    )
+    return vals
+
+
+@partial(jax.jit, static_argnames=("interp",))
+def _draw_with_interpolant_xval(x, y, xmin, ymin, zp, interp):
```

Comment:
@mrb This function and `_draw_with_interpolant_kval` are complicated and I don't really follow it. Could you give me some pointers (add a few comments to the functions) or point me to the corresponding c++ code so I can parse it? Thanks!

Reply:
done!